### PR TITLE
[JENKINS-30561] Attempt to install JDK if min java requirements are not met

### PR DESCRIPTION
**Pull request by e-g-hategan**
https://issues.jenkins-ci.org/browse/JENKINS-30561

### CODE REVIEW

**Comment by e-g-hategan** [comment]
Just clarifying: do you mean I should bump jenkins.version to latest 2.50, java.level to 7 and address any upgrade related issues?

**Comment by oleg-nenashev** [comment]
Definitely not to `2.50`. I am not sure what should be used exactly. `credentials` plugin dependency in the plugin already requires 1.625.3, hence I doubt this plugin actually can support Java6 anyway (CC @daniel-beck, @olivergondza , @stephenc since they are involved into the related changes).

So I suppose `1.625.3` is the best possible target

File: src/main/java/hudson/plugins/sshslaves/SSHLauncher.java [hunk 1]
```diff
@@ -1176,6 +1178,10 @@ protected String checkJavaVersion(final PrintStream logger, String javaCommand,
         return null;
     }
 
+    private double requiredJavaVersion() {
+        return Double.valueOf("1." + JDK_MAJOR_VERSION);
```
Comment (olivergondza) [inline src/main/java/hudson/plugins/sshslaves/SSHLauncher.java]: How about base this on a minimal version that is supported by jenkins version running on master (provided we can get that requirement at runtime)? Or at least use the version running on master as the minimum.

Comment (e-g-hategan) [inline src/main/java/hudson/plugins/sshslaves/SSHLauncher.java]: That's a good idea.

Comment (oleg-nenashev) [inline src/main/java/hudson/plugins/sshslaves/SSHLauncher.java]: It can be hardcoded by checking the remoting version

Comment (e-g-hategan) [inline src/main/java/hudson/plugins/sshslaves/SSHLauncher.java]: Based on Jenkins.getInstance().getVersion() we have 3 possible outcomes: 
before 1.610 then java 6
before 2.5x: java 7
afterwards java 8 (will happen soon; confirmed on the IRC channel).

Comment (oleg-nenashev) [inline src/main/java/hudson/plugins/sshslaves/SSHLauncher.java]: Something like that. But I am not sure this plugin can wok on Java 6 due to the credentials dependency

Comment (monkz) [inline src/main/java/hudson/plugins/sshslaves/SSHLauncher.java]: as jenkins.io offers a jdk list - i recommend using this list to choose a JDK version

**Pull request closed without merging**